File: phase-03/frontend/src/types/mapping.ts
/**
 * Display Mapping Types for AI Task Display Mapping Feature
 *
 * These types support the runtime mapping between display indices (user-facing)
 * and internal task IDs (database).
 */

/**
 * Single display mapping entry linking a display index to a task ID
 */
export interface DisplayMapping {
  display_index: number;  // 1-based sequential index shown to user
  task_id: string | number;  // Internal task identifier
}

/**
 * Complete display mapping state for a user session
 */
export interface DisplayMappingState {
  mappings: DisplayMapping[];
  user_id: string | number;
  last_updated: Date;
  isValid: boolean;
}

/**
 * Response from mapping refresh endpoint
 */
export interface MappingRefreshResponse {
  mapping_updated: boolean;
  total_mappings: number;
  display_mapping: DisplayMapping[];
  refreshed_at: string;  // ISO 8601 timestamp
}

/**
 * Request to resolve display index to task ID
 */
export interface ResolveDisplayIndexRequest {
  display_index: number;
  user_id: string | number;
}

/**
 * Response from display index resolution
 */
export interface ResolveDisplayIndexResponse {
  task_id: string | number;
  display_index: number;
  valid: boolean;
  error?: string;
}

/**
 * Request to refresh display mapping
 */
export interface RefreshMappingRequest {
  user_id: string | number;
  tasks: Array<{
    id: string | number;
    title: string;
    completed: boolean;
  }>;
}

File: phase-03/backend/src/schemas/mapping.py
"""
Display Mapping Schemas for AI Task Display Mapping Feature

These Pydantic schemas support the runtime mapping between display indices
(user-facing) and internal task IDs (database).
"""
from pydantic import BaseModel, Field
from typing import Union, Optional, List
from datetime import datetime


class DisplayMapping(BaseModel):
    """Single display mapping entry linking a display index to a task ID"""
    display_index: int = Field(..., ge=1, description="1-based sequential index shown to user")
    task_id: Union[str, int] = Field(..., description="Internal task identifier")


class DisplayMappingState(BaseModel):
    """Complete display mapping state for a user session"""
    mappings: List[DisplayMapping]
    user_id: Union[str, int]
    last_updated: datetime
    is_valid: bool = Field(default=True)


class MappingRefreshRequest(BaseModel):
    """Request to refresh display mapping"""
    user_id: Union[str, int] = Field(..., description="User identifier for mapping context")
    tasks: List[dict] = Field(..., description="List of task objects with id, title, completed")


class MappingRefreshResponse(BaseModel):
    """Response from mapping refresh endpoint"""
    mapping_updated: bool
    total_mappings: int
    display_mapping: List[DisplayMapping]
    refreshed_at: datetime


class ResolveDisplayIndexRequest(BaseModel):
    """Request to resolve display index to task ID"""
    display_index: int = Field(..., ge=1, description="Display index to resolve (1-based)")
    user_id: Union[str, int] = Field(..., description="User identifier")


class ResolveDisplayIndexResponse(BaseModel):
    """Response from display index resolution"""
    task_id: Union[str, int]
    display_index: int
    valid: bool
    error: Optional[str] = None
